src/shared/lib/validation/file.ts

```diff
@@ -1,5 +1,44 @@
-import { validateGenericString, Validation } from 'shared/lib/validation';
+import { get } from 'immutable';
+import { FilePermissions } from 'shared/lib/resources/file';
+import { parseUserType, UserType } from 'shared/lib/resources/user';
+import { adt, Id } from 'shared/lib/types';
+import { invalid, isValid, mapValid, valid, validateGenericString, validateUUID, Validation } from 'shared/lib/validation';
+import { isString } from 'util';
 
-export function validateFileName(name: string): Validation<string> {
-  return validateGenericString(name, 'File name');
+export function validateFileName(name: string, validExtensions: string[] = []): Validation<string> {
+  const validatedName = validateGenericString(name, 'File name');
+  if (isValid(validatedName) && validExtensions.length > 0) {
+    const extension = name.substr(name.lastIndexOf('.') + 1);
```

Comment:
Change `name` to `validatedName.value`. Also, what if the name doesn't contain a `'.'`?

Might be safer to use a regexp:

```ts
//...
const extension = name.match(/\.([a-zA-Z0-9]+)$/);
if (extension && validExtensions.map(ext => ext.toLowerCase()).includes(extension[1].toLowerCase())) {
//...
```